Spigot-Server-Patches/0076-Optimized-Light-Level-Comparisons.patch

```diff
@@ -64,8 +64,57 @@ index d0fa96651..62ec2ad31 100644
                              // CraftBukkit start
                              // world.setTypeUpdate(blockposition1, Blocks.GRASS.getBlockData());
                              org.bukkit.World bworld = world.getWorld();
+diff --git a/src/main/java/net/minecraft/server/BlockIceFrost.java b/src/main/java/net/minecraft/server/BlockIceFrost.java
+index 950863fb..4b04742a 100644
+--- a/src/main/java/net/minecraft/server/BlockIceFrost.java
++++ b/src/main/java/net/minecraft/server/BlockIceFrost.java
+@@ -19,7 +19,7 @@ public class BlockIceFrost extends BlockIce {
+     }
+ 
+     public void b(World world, BlockPosition blockposition, IBlockData iblockdata, Random random) {
+-        if ((random.nextInt(3) == 0 || this.c(world, blockposition) < 4) && world.getLightLevel(blockposition) > 11 - ((Integer) iblockdata.get(BlockIceFrost.a)).intValue() - iblockdata.c()) {
++        if ((random.nextInt(3) == 0 || this.c(world, blockposition) < 4) && world.compareLightLevel(blockposition, 11 - ((Integer) iblockdata.get(BlockIceFrost.a)).intValue() - iblockdata.c()) == 1) {
+             this.a(world, blockposition, iblockdata, random, true);
+         } else {
+             world.a(blockposition, (Block) this, MathHelper.nextInt(random, 20, 40));
+diff --git a/src/main/java/net/minecraft/server/BlockMycel.java b/src/main/java/net/minecraft/server/BlockMycel.java
+index 9461b4a2..084b2ec9 100644
+--- a/src/main/java/net/minecraft/server/BlockMycel.java
++++ b/src/main/java/net/minecraft/server/BlockMycel.java
+@@ -28,7 +28,8 @@ public class BlockMycel extends Block {
+ 
+     public void b(World world, BlockPosition blockposition, IBlockData iblockdata, Random random) {
+         if (!world.isClientSide) {
+-            if (world.getLightLevel(blockposition.up()) < 4 && world.getType(blockposition.up()).c() > 2) {
++            int lightLevel = -1; // Paper
++            if (world.getType(blockposition.up()).c() > 2 && (lightLevel = world.getLightLevel(blockposition.up())) < 4) { // Paper - move light check to end to avoid unneeded light lookups
```

Comment:
Instead of doing all this, maybe we could instead use a last access cache for light levels like we do for chunk lookups?

so if 2 things call getLightLevel in a row with same pos, 2nd is cached.

seems like a lot of diff to maintain here.